libdnf/transaction/Swdb.cpp

```diff
@@ -55,22 +57,52 @@ Swdb::Swdb(const std::string &path)
   , autoClose(true)
 {
     if (path == ":memory:") {
-        // writing to an in-memory database
-        conn = std::make_shared< SQLite3 >(path);
+        // connect to an in-memory database as requested
+        conn = std::make_shared<SQLite3>(path);
         Transformer::createDatabase(conn);
-    } else if (!pathExists(path.c_str())) {
-        // writing to a file that doesn't exist and must be created
-
-        // extract persistdir from path - "/var/lib/dnf/"
-        auto found = path.find_last_of("/");
-
-        Transformer transformer(path.substr(0, found), path);
-        transformer.transform();
-
-        conn = std::make_shared< SQLite3 >(path);
+    } else if (pathExists(path.c_str())) {
+        try {
+            // connect to an existing database
+            conn = std::make_shared<SQLite3>(path);
+            // try to execute a select to detect if the database is not read-only
+            conn->exec("BEGIN; UPDATE config SET value='test' WHERE key='test'; ROLLBACK;");
+        } catch (std::exception & ex) {
```

Comment:
... also you're catching all the errors, but you probably only want to catch permission errors, right? For example a corrupted database error you still want to propagate to the user technically speaking? Especially since you're ignoring the error silently...

Reply:
We want to catch *all* errors and propagate the error message to the user (see the log message).